### PR TITLE
chore(git): Run a quick linting task on pre-push

### LINKED ISSUE #2592: Run lint scripts on a prepush git hook
(From https://github.com/mozilla/fxa-content-server/pull/2591#issuecomment-111643568)

We could use something like https://www.npmjs.com/package/pre-commit or https://www.npmjs.com/package/husky and make it a devDep so everybody is forced into the precommit route.

IIRC, we used [**husky**](https://github.com/typicode/husky) (with [**grunt-newer**](https://github.com/tschaub/grunt-newer) so it'd only lint modified files instead of all files) in [mozilla/chronicle](https://github.com/mozilla/chronicle/) and then just ran the custom lint task on `prepush`.

### PR DESCRIPTION
**Pull request by eoger**
Fixes #2592

### CODE REVIEW

File: grunttasks/lint.js [hunk 1]
```diff
@@ -5,11 +5,17 @@
...
     'jsonlint:app',
     'jscs',
     'amdcheck'
-  ]);
+  ];
+
+  grunt.registerTask('lint', 'lint all the things', SUBTASKS);
+  grunt.registerTask('quicklint', 'lint the modified files', SUBTASKS.map(function (task) {
+    return 'newer:' + task;
+  }));
```
Comment (pdehaan) [inline grunttasks/lint.js]: SO PRETTY!

Comment (eoger) [inline grunttasks/lint.js]: Thank you [Chronicle](https://github.com/mozilla/chronicle/blob/master/grunttasks/lint.js) :)

**Comment by pdehaan** [comment]
Looks good. It seemed to work well for us in Chronicle, so we can give it a try and see if it works.

r? @zaach @vladikoff @shane-tomlinson

**Comment by vladikoff** [comment]
@eoger looking at this output: 

```
➜  fxa-content-server git:(issue-2592-prepush-lint) git push vladikoff issue-2592-prepush-lint 
Running "newer:eslint" (newer) task

Running "newer:eslint:grunt" (newer) task

Running "eslint:grunt" (eslint) task


Running "newer-postrun:eslint:grunt:1:/Users/vladikoff/mozilla/fxa-local-dev/fxa-content-server/node_modules/grunt-newer/.cache" (newer-postrun) task

Running "newer:eslint:app" (newer) task
No newer files to process.

Running "newer:eslint:tests" (newer) task
No newer files to process.

Running "newer:eslint:scripts" (newer) task
No newer files to process.

Running "newer:eslint:server" (newer) task
No newer files to process.

Running "newer:jshint" (newer) task

Running "newer:jshint:grunt" (newer) task

Running "jshint:grunt" (jshint) task

✔ No problems
Running "newer-postrun:jshint:grunt:2:/Users/vladikoff/mozilla/fxa-local-dev/fxa-content-server/node_modules/grunt-newer/.cache" (newer-postrun) task

Running "newer:jshint:app" (newer) task
No newer files to process.

Running "newer:jshint:tests" (newer) task
No newer files to process.

Running "newer:jshint:server" (newer) task
No newer files to process.

Running "newer:jsonlint:app" (newer) task
No newer files to process.

Running "newer:jscs" (newer) task

Running "newer:jscs:server" (newer) task
No newer files to process.

Running "newer:jscs:app" (newer) task
No newer files to process.

Running "newer:amdcheck" (newer) task

Running "newer:amdcheck:app" (newer) task
No newer files to process.
```

Does it have to show the .cache part? :
```

Running "newer-postrun:jshint:grunt:2:/Users/vladikoff/mozilla/fxa-local-dev/fxa-content-server/node_modules/grunt-newer/.cache" (newer-postrun) task
```

**Comment by vladikoff** [comment]
Otherwise seems to work well!

**Comment by pdehaan** [comment]
@vladikoff Does it always reproduce, or only the first time when it builds the initial cache?
I vaguely recall poking at this pre-Whistler and noticed my first precommit was slow since I didn't have the cache, but the second time it was pretty quick.

**Comment by vladikoff** [comment]
If we can improve this by excluding certain directories from `grunt-newer` then please let me know.

**Pull request merged**